src/main/java/games/strategy/triplea/ui/TripleAFrame.java

```diff
@@ -924,6 +929,37 @@ public Unit getStrategicBombingRaidTarget(final Territory territory, final Colle
     return selected.get();
   }
 
+  @SuppressWarnings("serial")
+  class UnitRenderer extends JLabel implements ListCellRenderer<Unit> {
+
+    public UnitRenderer() {
+      setOpaque(true);
+    }
+
+    @Override
+    public Component getListCellRendererComponent(final JList<? extends Unit> list, final Unit unit, final int index,
```

Comment:
A quick comment here might be good, summarize that we are building a dialog with the unit image next to name.

Reply:
Added comment to class